src/EditorFeatures/CSharp/CommentSelection/ToggleBlockCommentCommandHandler.cs

```diff
@@ -0,0 +1,428 @@
+    {
+        [ImportingConstructor]
+        internal ToggleBlockCommentCommandHandler(
+            ITextUndoHistoryRegistry undoHistoryRegistry,
+            IEditorOperationsFactoryService editorOperationsFactoryService) : base(undoHistoryRegistry, editorOperationsFactoryService)
+        { }
+
+        /* TODO - modify once the toggle block comment handler is added.
+        public VSCommanding.CommandState GetCommandState(CommentSelectionCommandArgs args)
+        {
+            return GetCommandState(args.SubjectBuffer);
+        }
+
+        public bool ExecuteCommand(CommentSelectionCommandArgs args, CommandExecutionContext context)
+        {
+            return ExecuteCommand(args.TextView, args.SubjectBuffer, Operation.Toggle, context);
+        }*/
+
+        public override string DisplayName => EditorFeaturesResources.Toggle_Block_Comment;
+
+        protected override string GetTitle(Operation operation)
+        {
+            return EditorFeaturesResources.Toggle_Block_Comment;
+        }
+
+        protected override string GetMessage(Operation operation)
+        {
+            return EditorFeaturesResources.Toggling_block_comment_on_selection;
+        }
+
+        protected override void SetTrackingSpans(ITextView textView, ITextBuffer buffer, List<CommentTrackingSpan> trackingSpans)
+        {
+            var spans = trackingSpans.Select(trackingSpan => trackingSpan.ToSelection(buffer));
+            textView.GetMultiSelectionBroker().SetSelectionRange(spans, spans.Last());
+        }
+
+        internal override void CollectEdits(Document document, ICommentSelectionService service, NormalizedSnapshotSpanCollection selectedSpans,
+            List<TextChange> textChanges, List<CommentTrackingSpan> trackingSpans, Operation operation, CancellationToken cancellationToken)
+        {
+            var experimentationService = document.Project.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
+            if (!experimentationService.IsExperimentEnabled(WellKnownExperimentNames.RoslynToggleBlockComment))
+            {
+                return;
+            }
+
+            if (selectedSpans.IsEmpty())
+            {
+                return;
+            }
+
+            var getRoot = document.GetSyntaxRootAsync();
+            var commentInfo = service.GetInfoAsync(document, selectedSpans.First().Span.ToTextSpan(), cancellationToken).WaitAndGetResult(cancellationToken);
+            var root = getRoot.WaitAndGetResult(cancellationToken);
+            if (commentInfo.SupportsBlockComment)
+            {
+                ToggleBlockComments(commentInfo, root, selectedSpans, textChanges, trackingSpans);
+            }
+        }
+
+        private static void ToggleBlockComments(CommentSelectionInfo commentInfo, SyntaxNode root, NormalizedSnapshotSpanCollection selectedSpans,
+            List<TextChange> textChanges, List<CommentTrackingSpan> trackingSpans)
```

Comment:
please try to avoid lines longer than github's width at 1080p (higest use resolution for VS devs).